uni/lib/model/providers/plausible/plausible_provider.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:async';
+
+import 'package:battery_plus/battery_plus.dart';
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:flutter/widgets.dart';
+import 'package:logger/logger.dart';
+import 'package:plausible_analytics/plausible_analytics.dart';
+import 'package:provider/provider.dart';
+import 'package:uni/controller/local_storage/preferences_controller.dart';
+
+class PlausibleProvider extends StatefulWidget {
+  const PlausibleProvider({
+    required this.plausible,
+    required this.child,
+    super.key,
+  });
+
+  final Plausible? plausible;
+  final Widget child;
+
+  @override
+  State<StatefulWidget> createState() => _PlausibleProviderState();
+}
+
+class _PlausibleProviderState extends State<PlausibleProvider> {
+  int _batteryLevel = 0;
+  bool _isInBatterySaveMode = true;
+  ConnectivityResult _connectivityResult = ConnectivityResult.mobile;
+  bool _isUsageStatsEnabled = true;
+
+  bool _canUpdateBatteryState = true;
+
+  @override
+  void initState() {
+    super.initState();
+
+    final plausible = widget.plausible;
+    if (plausible != null) {
+      plausible.enabled = false;
+
+      _startListeners(plausible)
+          .then((_) => _updateBatteryState())
+          .then((_) => _updateConnectivityState())
+          .then((_) => _updateUsageStatsState());
+    }
+  }
+
+  void _updateAnalyticsState() {
+    final plausible = widget.plausible;
+    if (plausible != null) {
+      plausible.enabled = _isUsageStatsEnabled &&
+          !_isInBatterySaveMode &&
+          _batteryLevel > 20 &&
+          _connectivityResult == ConnectivityResult.wifi;
+
+      Logger().d('Plausible enabled: ${plausible.enabled}');
+    }
+  }
+
+  Future<void> _updateBatteryState() async {
+    if (!_canUpdateBatteryState) {
+      return;
+    }
+
+    _canUpdateBatteryState = false;
+    Timer(const Duration(seconds: 10), () => _canUpdateBatteryState = true);
```

Comment:
Maybe more than 10 seconds is more relaxed?

Reply:
Yes, it was originally 1 minute. I changed it to 10 seconds for testing and forgot to change it back. Does 1min sound good to you?

Reply:
Sure, sounds good